src/stepfunctions/inputs/placeholders.py

```diff
@@ -291,3 +291,50 @@ def _create_variable(self, name, parent, type=None):
             return StepInput(name=name, parent=parent, type=type)
         else:
             return StepInput(name=name, parent=parent)
+
+
+class MapItemValue(Placeholder):
+    """
+        Top-level class for map item value placeholders.
+    """
+
+    def __init__(self, schema=None, **kwargs):
+        super(MapItemValue, self).__init__(schema, **kwargs)
+        self.json_str_template = '$$.Map.Item.Value{}'
+
+    def _create_variable(self, name, parent, type=None):
+        """
+            Creates a placeholder variable for Map Item Value.
+            A placeholder variable can only be created if the collection is not immutable due to a pre-specified schema.
+        """
+        if self.immutable:
+            raise ValueError("Placeholder variable does not conform to schema set for the placeholder collection.")
+        if type:
+            return MapItemValue(name=name, parent=parent, type=type)
+        else:
+            return MapItemValue(name=name, parent=parent)
+
+
+class MapItemIndex(Placeholder):
+    """
+        Top-level class for map item index placeholders.
+    """
+
+    def __init__(self, schema=None, **kwargs):
```

Comment:
question: is there a test that uses / sets this Schema

Reply:
Yes, there is - test_map_item_value_with_schema() in tests/unit/test_placeholders.py uses the schema
We validate the schema and the exceptions that are thrown if the schema is not respected
I can add an assert for success as well

Reply:
Just read this again and realizing this is for `MapItemIndex` and not `MapItemValue`
The `MapItemIndex` should not have a schema as it can only return the index of the map iteration - I will make the change in the next commit